@xen-orchestra/web-core/lib/components/DonutChart.vue

```diff
@@ -0,0 +1,162 @@
+<template>
+  <div class="wrapper">
+    <div class="donut-chart" :style="gradientStyle">
+      <span class="donut-hole"><UiIcon v-if="icon" :icon="icon" class="chart-icon" /></span>
+    </div>
+
+    <ul class="chart-legends">
+      <h3 v-if="title" class="chart-title">{{ title }}</h3>
+      <li class="legend" v-for="legend in chartFigures" :key="legend.legend">
+        <span class="legend-dot" :class="legend.color"></span><span class="legend-text">{{ legend.legend }}</span
+        ><span class="legend-value">{{ legend.value }}</span>
+      </li>
+    </ul>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed } from 'vue'
+import UiIcon from '@core/components/icon/UiIcon.vue'
+import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
+
+type ChartFigure = {
+  value: number
+  color: string
+  legend: string
+}
+
+interface Props {
+  chartFigures: ChartFigure[]
+  title?: string
+  icon?: IconDefinition
+}
+
+const props = withDefaults(defineProps<Props>(), {
+  chartFigures: () => [
+    {
+      value: 1,
+      color: 'unknown',
+      legend: 'Unknown',
+    },
+  ],
+})
+
+const totalValue = props.chartFigures.reduce((acc, chartFigure) => acc + chartFigure.value, 0)
+
+function convertColorToVar(color: string) {
+  switch (color) {
+    case 'info':
+      return 'var(--color-purple-base)'
+    case 'success':
+      return 'var(--color-green-base)'
+    case 'warning':
+      return 'var(--color-orange-base)'
+    case 'error':
+      return 'var(--color-red-base)'
+    case 'unknown':
+      return 'var(--background-color-purple-10)'
+    default:
+      return 'var(--background-color-primary)'
+  }
+}
+
+const gradientStyle = computed(() => {
+  let cumulativePercent = 0
+  let gradientStr = 'background: conic-gradient('
+  props.chartFigures.forEach((figure, index) => {
+    const nextPercent = (figure.value / totalValue) * 100
+    if (index > 0) {
+      gradientStr += `, ${convertColorToVar(figure.color)} ${cumulativePercent}% ${cumulativePercent + nextPercent}%`
+    } else {
+      gradientStr += `${convertColorToVar(figure.color)} 0% ${nextPercent}%`
+    }
+    cumulativePercent += nextPercent
+  })
+  gradientStr += ')'
+  return gradientStr
+})
```

Comment:
ℹ️ This can be simplified

```suggestion
const gradientStyle = computed(() => {
  let cumulativePercent = 0
  const parts = props.chartFigures.map(figure => {
    const currentPercent = cumulativePercent
    cumulativePercent += (figure.value / totalValue) * 100
    return `${convertColorToVar(figure.color)} ${currentPercent}% ${cumulativePercent}%`
  })

  return { background: `conic-gradient(${parts.join(', ')}` }
})
```